vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/view/components/ProjectLinksComponent.java

```diff
@@ -48,30 +57,69 @@ public ProjectLinksComponent(@Autowired ProjectInformationService projectInforma
           button.addThemeVariants(ButtonVariant.LUMO_ICON,
               ButtonVariant.LUMO_ERROR,
               ButtonVariant.LUMO_TERTIARY);
-          button.addClickListener(e -> {
-            linkList.remove(projectLink);
-            projectLinks.getDataProvider().refreshAll();
-            }
-          );
-          button.setIcon(new Icon("lumo","cross"));
+          button.setIcon(new Icon("lumo", "cross"));
+          button.addClickListener(e -> removeLink(projectLink));
         })
     );
-    projectLinks.setItems(linkList);
+    projectLinks.setItems(new ArrayList<>());
+    offerSearchComponent.addSelectedOfferChangeListener(it -> {
+      if (Objects.isNull(it.getValue())) {
+        return;
+      }
+      if (!it.isFromClient()) {
+        return;
```

Comment:
What's the point of this check? Can the selected Offer not only be generated from the client in this context?

Reply:
When we reset the value in the value change listener, we want to ignore the event created by that. As `clear` calls `setValue(emptyValue())` a change event is fired from the server.

Reply:
Aaah that makes sense thanks for explaining 🧠